test/integration/webhookgithub_test.go

```diff
@@ -299,6 +300,11 @@ func postFile(client restclient.HTTPClient, event, filename, url string, expStat
 	if resp.StatusCode != expStatusCode {
 		t.Errorf("Wrong response code, expecting %d, got %d: %s!", expStatusCode, resp.StatusCode, string(body))
 	}
+	var newBuild buildapi.Build
+	if err = runtime.DecodeInto(kapi.Codecs.UniversalDecoder(), body, &newBuild); err != nil {
```

Comment:
I believe @deads2k was suggesting you actually inspect the build object to ensure it contains the content the test would expect it to contain.  (validate a couple of the fields like name, strategy, source)